### ISSUE TITLE
Impossible to delete the seed

### ISSUE THREAD

**Issue by kayrus** [gardener/gardener#1352]
**What happened**:

Seed cannot be deleted. I executed the following command:

```
kubectl delete seed region1-01-dev region2-01-dev
```

Only region2-01-dev was removed. The region1-01-dev seed hangs with the following messages:

```
time="2019-08-20T08:50:26Z" level=info msg="Error syncing ControllerInstallation provider-openstack-zfbxm: secrets \"seed-region1-01-dev\" not found"
time="2019-08-20T08:50:27Z" level=info msg="Error syncing ControllerInstallation provider-openstack-zfbxm: secrets \"seed-region1-01-dev\" not found"
time="2019-08-20T08:50:27Z" level=info msg="Error syncing ControllerInstallation provider-openstack-zfbxm: secrets \"seed-region1-01-dev\" not found"
```

kubectl debug:

```
$ kubectl describe ControllerInstallation provider-openstack-zfbxm
Name:         provider-openstack-zfbxm
Namespace:    
Labels:       registration-spec-hash=842fd387167b39bd
              seed-spec-hash=4fc26771b078f3ee
Annotations:  <none>
API Version:  core.gardener.cloud/v1alpha1
Kind:         ControllerInstallation
Metadata:
  Creation Timestamp:             2019-08-20T08:16:49Z
  Deletion Grace Period Seconds:  0
  Deletion Timestamp:             2019-08-20T08:35:51Z
  Finalizers:
    core.gardener.cloud/controllerinstallation
  Generate Name:     provider-openstack-
  Generation:        17
  Resource Version:  573831
  Self Link:         /apis/core.gardener.cloud/v1alpha1/controllerinstallations/provider-openstack-zfbxm
  UID:               dbb4a34d-c322-11e9-b854-0a58645f0007
Spec:
  Registration Ref:
    Name:              provider-openstack
    Resource Version:  464166
  Seed Ref:
    Name:              region1-01-dev
    Resource Version:  571486
Status:
  Conditions:
    Last Transition Time:  2019-08-20T08:50:27Z
    Last Update Time:      2019-08-20T08:50:27Z
    Message:               Referenced Seed does not exist: secrets "seed-region1-01-dev" not found
    Reason:                SeedNotFound
    Status:                False
    Type:                  Valid
    Last Transition Time:  2019-08-20T08:50:27Z
    Last Update Time:      <nil>
    Message:               The condition has been initialized but its semantic check has not been performed yet.
    Reason:                ConditionInitialized
    Status:                Unknown
    Type:                  Installed
  Provider Status:
    Resources:
      API Version:  v1
      Kind:         ConfigMap
      Name:         gardener-extension-provider-openstack-configmap
      Namespace:    extension-provider-openstack-zfbxm
      API Version:  storage.k8s.io/v1
      Kind:         StorageClass
      Name:         gardener.cloud-fast
      API Version:  v1
      Kind:         ServiceAccount
      Name:         gardener-extension-provider-openstack
      Namespace:    extension-provider-openstack-zfbxm
      API Version:  rbac.authorization.k8s.io/v1
      Kind:         ClusterRole
      Name:         gardener-extension-provider-openstack
      API Version:  rbac.authorization.k8s.io/v1
      Kind:         ClusterRoleBinding
      Name:         gardener-extension-provider-openstack
      API Version:  v1
      Kind:         Service
      Name:         gardener-extension-provider-openstack
      Namespace:    extension-provider-openstack-zfbxm
      API Version:  apps/v1
      Kind:         Deployment
      Name:         gardener-extension-provider-openstack
      Namespace:    extension-provider-openstack-zfbxm
      API Version:  autoscaling.k8s.io/v1beta2
      Kind:         VerticalPodAutoscaler
      Name:         gardener-extension-provider-openstack-vpa
      Namespace:    extension-provider-openstack-zfbxm
Events:             <none>
```

**Environment**:

- Gardener version: 0.27.6
- Kubernetes version (use `kubectl version`): 1.15.0
- Cloud provider or hardware configuration: openstack
- Others: gardener-extensions to 0.9.3

**Comment by kayrus** [comment]
Looks like a race condition. Region2 secrets were not removed, but region1 secrets were removed:

```
$ kubectl -n garden get secret | grep seed
root-dev-seed                                                 Opaque                                7      2d23h
seed-region2-01-dev                                           Opaque                                6      17h
seed-soil-dev                                                 Opaque                                6      2d22h
```

I never faced this issue when I removed seeds one by one. It happened only when I specified two of them in one command.

**Comment by kayrus** [comment]
Just started to cleanup the seeds and shoots for the new clean deployment and noticed this:

```
$ kubectl -n garden get secrets
NAME                  TYPE     DATA   AGE
seed-region1-01-dev   Opaque   6      28h
seed-region2-01-dev   Opaque   6      45h
seed-region3-01-dev   Opaque   6      45h
seed-soil-dev         Opaque   6      4d2h
```

Not sure whether it is related. In the initial issue the `seed-region1-01-dev` was removed before removing the seed, in this situation seeds were removed, but not the secrets.

**Comment by rfranzke** [comment]
I think this should be fixed with the introduction of the Gardenlet #1601. Please reopen if you think it's not.
